addons/mail/static/src/components/attachment_box/attachment_box.xml

```diff
@@ -3,7 +3,7 @@
 
     <t t-name="mail.AttachmentBox" owl="1">
         <div class="o_AttachmentBox">
-            <t t-if="chatter">
+            <t t-if="attachmentBoxView.chatter">
```

Comment:
guard should be on the top level record of the component, so only `attachmentBoxView`.

`chatter` is a required field so it will always be set.